hal/src/main/native/sim/HAL.cpp

```diff
@@ -193,7 +201,68 @@ HAL_Bool HAL_GetBrownedOut(int32_t* status) {
   return false;  // Figure out if we need to detect a brownout condition
 }
 
+/* A return < 0 indicates an error that should stop the HAL.
+   0 is success, and > 0 is a non fatal error */
+typedef int halsim_init_func_t(void);
+
+int HAL_LoadExtraSimulation(const char* library) {
+  int rc = 1;  // It is expected and reasonable not to find an extra simulation
+#if defined(WIN32) || defined(_WIN32)
+  HMODULE handle = LoadLibrary(library);
+  if (!handle) {
+    std::string library_name = std::string(library) + ".dll";
+    handle = LoadLibrary(library_name.c_str());
+  }
+#else
+  void* handle = dlopen(library, RTLD_LAZY);
+  if (!handle) {
+    std::string library_name = "lib" + std::string(library) + ".so";
+    handle = dlopen(library_name.c_str(), RTLD_LAZY);
+  }
+#endif
+  if (!handle) return rc;
+
+  halsim_init_func_t* init;
+#if defined(WIN32) || defined(_WIN32)
+  init = reinterpret_cast<halsim_init_func_t*>(GetProcAddress(handle, "init"));
```

Comment:
Maybe rename these to HAL_SimInit as the symbol name? `init` could be ambiguous for sim libraries that carry a library along with them.

Reply:
Yes, that's a god point.